packages/manager/apps/pci-public-ip/src/pages/list/ListGuard.tsx

```diff
@@ -21,12 +23,15 @@ export default function ListGuard({
     isLoading: isFloatingIPsLoading,
   } = useAllFloatingIP(projectId);
 
-  if (!isFailoverIPsLoading && !isFloatingIPsLoading) {
-    if (failoverIPs?.length > 0 || floatingIPs?.length > 0) {
-      return children;
+  useEffect(() => {
+    if (!isFailoverIPsLoading && !isFloatingIPsLoading) {
+      if (failoverIPs?.length > 0 || floatingIPs?.length > 0) {
+        setView(children);
+      } else {
+        navigate(`/pci/projects/${projectId}/public-ips/onboarding`);
+      }
     }
-    navigate('../onboarding');
-  }
+  }, [navigate, failoverIPs, floatingIPs]);
```

Comment:
Maybe add the both loading in the useEffect dependencies ?

Reply:
done